_posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc

```diff
@@ -763,372 +763,2409 @@ Terraform::
+your credentials) or for the
+https://docs.aws.amazon.com/general/latest/gr/aws_tasks-that-require-root.html[small number of tasks that require root user credentials].
+
+
+[[create_iam_user_in_root]]
+=== Create an IAM user in the root account
+
+As the last action you do as the root user, you MUST create an IAM user. This is not only a better practice from a
+security standpoint, but also, the `account-baseline-xxx` modules we will use below assume IAM roles, which does not
+work with a root user. Later on, we'll create and manage all IAM users as code, but you should create this very first
+IAM user manually by
+https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html#id_users_create_console[following these instructions]:
+
+. Enter a username for your IAM user.
+. Select both "programmatic access" and "AWS Management Console access."
+. On the next page, click "Attach existing policies to user directly" and attach the `AdministratorAccess` policy.
+. Click next a few more times to create the IAM user.
+. In a secrets manager, save the IAM sign-in URL, your IAM user's username, the password, and your Access Keys.
+
+[[lock_down_iam_users]]
+=== Lock down the root account IAM users
+
+Although IAM users don't have the same powers as a root user, having an IAM user account compromised can still be a
+huge problem for your company (especially if that IAM user had admin permissions), so it's still critical to lock down
+IAM user accounts as much as possible:
+
+Use a secrets manager::
+Do NOT store the root user's password, or secrets of any kind, in plain text. Instead, always use a secrets manager
+such as https://1password.com[1Password], https://www.lastpass.com[LastPass], or https://www.passwordstore.org[pass]
+to store the credentials in an encrypted format.
+
+
+Use a strong, generated password::
+Do NOT re-use passwords from other websites, or any password that you can remember at all. Instead, generate a random,
+cryptographically secure, long password (20+ characters). All the password managers mentioned above can generate and
+store passwords for you in one step, so use them!
+
+
+Enable MFA::
+Always make sure to
+https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_mfa_enable.html[enable MFA for your IAM user].
+Feel free to use a virtual or hardware MFA device—whichever is easier or required by your company—as either one
+dramatically improves the security of your IAM user. Note that using SMS (text messages) for MFA is
+https://www.schneier.com/blog/archives/2016/08/nist_is_no_long.html[no longer recommended by NIST] due to known
+https://www.theverge.com/2017/9/18/16328172/sms-two-factor-authentication-hack-password-bitcoin[vulnerabilities with the cellular system],
+so using a virtual or hardware MFA device is preferable; that said, MFA with SMS is still better than no MFA at all.
+
+Complete this step before creating any IAM users!
```

Comment:
This sentence is confusing - which step is it referring to? Could we clarify this a bit?

Reply:
you're right - it's a bit out of nowhere.... not sure what it was referring to, but now I'd vote let's remove this line for now.

Reply:
I'm removing for now!